.github/workflows/build-test-lint.yml

```diff
@@ -0,0 +1,87 @@
+name: "Build, Test, Lint"
+on: [push, pull_request]
+jobs:
+  build:
+    strategy:
+      matrix:
```

Comment:
this 1*1*1 matrix seems not necessary... unless it is prep for future multi-version support?

Reply:
We do target amd64 and arm64 for release, maybe we might also need to test arm64 in the future part of CI